src/mame/drivers/prodigy.cpp

```diff
@@ -546,7 +546,7 @@ void prodigy_state::update_bcd()
 		case 0x02: digit_nbr = 1; break;
 		case 0x04: digit_nbr = 2; break;
 		case 0x08: digit_nbr = 3; break;
-		default: logerror("Wrong BCD digit, shouldn't happen, call the maintainer!\n");
+		default: fatalerror("Wrong BCD digit, shouldn't happen, call the maintainer!\n");
```

Comment:
Please don't do this - it's not bad enough to be a fatal error and will make development harder.

Reply:
The problem is, that digit_nbr is initialized to 4 which will cause a out-of-bounds access when it accesses ```uint8_t m_digit_cache[4]``` a few lines down (which is what Coverity complained about).

Reply:
You can remove default: altogether. It's impossible(both on the real chip and this emulation) for (ttl74145_data & 0x0f) to be a different value than 0,1,2,4,8.

Then, simply change:
uint8_t digit_nbr = 4;
to: uint8_t digit_nbr = 0;

Compiler and/or test tools will probably complain too if left unitialized.